src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -4212,9 +4180,9 @@ public static String getMessage4(
 
     public static final MessageProvider messageProvider = new DefaultMessageProvider();
 
-    public static String getMessage(String messageId, Object[] arguments)
+    public static String getMessage(String messageId, Object... args)
```

Comment:
Sadly here I am also sure that this will break someone, but if you instead deprecated the old one and created a new one then we could modernize the codebase while still being safe.